### DISCUSSION TITLE
[Config Support]: Validating config before saving

### DISCUSSION THREAD

**Discussion by F1VWdk** [blakeblackshear/frigate#13044] [Config Support]
### Describe the problem you are having

Sorry for this, maybe simple and obvious, question but I just cant find any additional information on how to implement validation of the config before saving. I've read the docs over and over staring at the section outlining how to do it but .... WHERE do I enter those lines of code? And how does it get triggered?

Any help would be appreciated, please.
Thank you

### Version

0.14

### Frigate config file

```yaml
n/a
```

### Relevant log output

```shell
n/a
```

### Operating system

HassOS

### Install method

HassOS Addon

### Object Detector

Coral

**Comment by NickM-27** [comment]
If you use the UI it always validates before saving

**Reply by F1VWdk** [reply]
I do use UI for what I can, but a lot of settings require config file changes. For example, I recently needed to change my alerts to be only limited to certain zones and the same limitation for mqtt and snapshots. So, i made config changes and added required_zones to global settings of record, mqtt, and snapshots by listing all zones from all cameras that I need which  caused Frigate not to start. I ended up moving required_zones for each setting into each camera section by listing only the zones of that each camera. It would've been nice if I knew how to validate the settings before derailing myself

**Reply by NickM-27** [reply]
By UI I meant the configuration editor in the UI

**Reply by F1VWdk** [reply]
yeah, UI = settings gear icon -> settings
config edit = settings gear icon - configuration editor

**Reply by NickM-27** [reply]
Right, any time that is saved it is validated.
